### PR TITLE
Add env prefix to Redis stream key

### PR DESCRIPTION
**Pull request by dbernard**

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+./setup.py test --test-suite tests
+```
+
+To run the integration tests:
+
+```shell
+dc up
+```
+
+(Note: The integration tests will display in the container output)
```
Comment (dbernard) [inline tests/README.md:15]: I suspect we can do some more intuitive "stuff" here, but I figured this was at least helpful for the time being.

File: feats/redis.py [hunk 2]
```diff
@@ -35,9 +37,20 @@ class FeatureStream:
...
-        self.key = f"feature:{key}"
+        self.key = self._get_key(key)
         self._redis = redis
 
+    def _get_key(self, key) -> str:
+        """
+        Builds the Redis stream key. If the `FEATS_ENV` environment variable is
+        set, it will be used as the prefix for the feature key.
+        """
+        env = os.getenv('FEATS_ENV')
```
Comment (chaten) [inline feats/redis.py]: What if we had this be an input to the RedisClient object instead? That way we aren't dependent on a global variable like the environment.

Comment (dbernard) [reply feats/redis.py]: I could just be misunderstanding the way feats interacts with the rest of the world, but would that mean we'd need to push different code to each environment? I.e. with `RedisClient(xyz, prefix='staging')` for staging, `RedisClient(xyz, prefix=None)` for prod, etc?

Comment (chaten) [reply feats/redis.py]: Different values would need to be passed, but the settings file knows which environment it is being deployed to, and can combine that with any environment variables we need. 

For example, our memcache setting grabs the key prefix from the environment and passes that through to the memcache client -- the client itself has no knowledge that it came from os.environ.

Comment (dbernard) [reply feats/redis.py]: 👍  I like it (completely forgot there was a settings file 😅 )

File: feats/redis.py [hunk 3]
```diff
@@ -103,7 +112,8 @@ class RedisClient:
     Initializing this class initializes the redis connection client but does
     _not_ test the connection to redis server.
     """
-    def __init__(self, host='localhost', port=6379, db=0, **options):
+    def __init__(self, host='localhost', port=6379, db=0, stream_prefix=None, **options):
```
Comment (chaten) [inline feats/redis.py]:
```suggestion
    def __init__(self, host='localhost', port=6379, db=0, key_prefix=None, **options):
```
We need any keys we create, regardless of if they are a stream or not, to be separate from other users.

**Review by chaten — APPROVED** [review]

**Pull request merged**